ttnn/cpp/ttnn/deprecated/tt_dnn/op_library/reduce/multi_core_h/reduce_op_multi_core_h.cpp

```diff
@@ -18,14 +18,23 @@ namespace tt {
 namespace tt_metal {
 
 operation::ProgramWithCallbacks reduce_multi_core_h(
-    const Tensor &a, Tensor &output, ReduceOpMath reduce_op, float scaler) {
+    const Tensor &a, Tensor &output, ReduceOpMath reduce_op, float scaler, const std::optional<DeviceComputeKernelConfig>& compute_kernel_config) {
     const auto shape = a.get_legacy_shape();
     uint32_t W = shape[3], H = shape[2], NC = shape[1] * shape[0];
 
     uint32_t Wt = W / TILE_WIDTH;
     uint32_t Ht = H / TILE_HEIGHT;
     uint32_t HtWt = Ht * Wt;
 
+    bool fp32_dest_acc_en = false;
+    MathFidelity math_fidelity = MathFidelity::HiFi4;
+
+    if (compute_kernel_config.has_value()) {
+        auto kernel_args = get_compute_kernel_config_args(a.device()->arch(), compute_kernel_config.value());
+        fp32_dest_acc_en = std::get<2>(kernel_args);
+        math_fidelity = std::get<0>(kernel_args);
+    }
```

Comment:
I think we can move this logic to top level `reduce` op code and always have `compute_kernel_config` to be populated and non-optional inside these program files.

Reply:
agreed

Reply:
ok